### PR TITLE
Json diff for top level `sys:JSONDocument`

### LINKED ISSUE #1394: Apply does not work on JSONDocuments at top level
The following does not work because the document complains that it can not infer the document type. This is because it *must* use a raw_json flag on insertion, which must likewise be reported by the diff. 

```shell
# Create Database
terminusdb db create admin/foo
# Add a JSON document
echo '{ "name": "Socrates" }' | terminusdb doc insert admin/foo -j
# Get the document back
terminusdb doc get admin/foo # Prove that it works, Socrates is there!
# Branch the database
terminusdb branch create admin/foo/local/branch/changes --origin admin/foo
# Add some more philosophers
echo '{ "name": "Plato" }' | terminusdb doc insert admin/foo/local/branch/changes -j
echo '{ "name": "Aristotle" }' | terminusdb doc insert admin/foo/local/branch/changes -j

# Look at the difference in the databases
terminusdb diff admin/foo --before-commit main --after-commit changes | jq

# Apply the differences

terminusdb apply admin/foo --before-commit main --after-commit changes | jq
```

### PR DESCRIPTION
**Pull request by GavinMendelGleason**
Fixes #1394

### CODE REVIEW

File: src/core/api/api_patch.pl [hunk 1]
```diff
@@ -64,7 +64,12 @@
         open_descriptor(Commit_Descriptor, commit_info{}, Transaction, Map_In, Map_Out),
         error(unresolvable_collection(Commit_Descriptor),_)),
 
-    get_document(Transaction, Doc_Id, Document).
+    Options = options{
+                  compress_ids : true,
+                  unfold: true,
+                  keep_json_type: true
```
Comment (matko) [inline src/core/api/api_patch.pl:70]: Should this also be implemented in the rust version?

File: src/core/document/json.pl [hunk 2]
```diff
@@ -570,6 +572,10 @@
             select_dict(json{'@id' : _ }, Context_With_ID, Context)
         ;   Context = _{})
     ).
+/*
+    default_prefixes(Prefixes),
+    put_dict(Context,Prefixes,Expanded).
+*/
```
Comment (matko) [inline src/core/document/json.pl:578]: might as well remove these comments

File: src/core/document/json.pl [hunk 3]
```diff
@@ -586,6 +592,11 @@
     create_context(Askable, Query_Context),
     database_prefixes(Query_Context, Context).
 
+database_and_default_prefixes(Askable, Prefixes) :-
+    database_prefixes(Askable, Database_Prefixes),
+    default_prefixes(Default_Prefixes),
```
Comment (matko) [inline src/core/document/json.pl:597]: if we're gonna have a set of default prefixes, we should probably enforce that those prefixes cannot be used by users in their own graphs. Do we do that?

Comment (GavinMendelGleason) [reply src/core/document/json.pl:597]: I don't think that is necessary. If someone uses a prefix to mean something else, and they submit something else, that's their problem not ours.

**Review by matko — APPROVED** [review]

**Pull request merged**